### PR TITLE
A bunch of stuff for the Travis CI

### PR DESCRIPTION
**Pull request by duncan3dc**
* General tidy up to make each section readable and follow standard community practices.
* Have travis run the unit tests (using phpunit)
* Use `composer update` as well as `composer install` as the lock file is irrelevant for users, they'll be using the latest supported dependencies (from `composer.json`)
* Also test against the lowest supported dependencies (from `composer.json`)
* Added testing on the latest supported php version (5.6) and the upcoming release (7.0)

Doing all this uncovered an [issue](https://travis-ci.org/duncan3dc/symfony-console-completion/builds/52403268) in the dependencies, where it was stated the phpunit 4.1 was required, when actually 4.4 is required. so I've bumped that requirement in `composer.json`

### CODE REVIEW

**Comment by stecman** [comment]
Thanks, @duncan3dc. I'm not sure about triggering [15 builds](https://travis-ci.org/stecman/symfony-console-completion/builds/52405254) though - seems a bit excessive. What do you think about just running `update` and `update --prefer-lowest`, or maybe only running `--prefer-lowest` for one PHP version (so six builds)?

**Comment by duncan3dc** [comment]
@stecman Yeh, I'm against having `composer.lock` in the repo generally, I only left the `composer install` because you had it. Shall I remove `composer.lock` as well?

**Comment by aik099** [comment]
I'm for `composer.lock` being committed because it explicitly allows for library developers to work using same dependency versions.

The `COMPOSER_OPTS` is a bit too much. Nobody really does that. In dependency testing the only needed stuff is:
1. test against latest stable versions
2. test against latest unstable versions

In general you can restrict matrix by checking all these unusual combinations for a single php version only.

**Comment by duncan3dc** [comment]
I'm not sure `'Nobody really does that'` is a relevant argument. By adding in this check an issue with the dependencies was actually found. If that doesn't state the value of this then I'm not sure what else could

**Comment by duncan3dc** [comment]
I've removed the `composer install` commands and the `composer.lock` file now.

**Comment by aik099** [comment]
My apologies for not backing up my `Nobody really does that` argument. Can you please reference well known packages/repos that use proposed Travis CI techniques.

Also it would be interesting to see these `standard community practices` explained somewhere in more detail, e.g. on Travis CI website or another as solid location.

File: composer.json [hunk 1]
```diff
@@ -13,7 +13,7 @@
         "symfony/console": "~2.2"
     },
     "require-dev": {
-        "phpunit/phpunit": "~4.1"
+        "phpunit/phpunit": "~4.4"
```
Comment (aik099) [inline composer.json]: Could you please reference what features of PHPUnit 4.4 are being used?

Comment (duncan3dc) [inline composer.json]: https://github.com/duncan3dc/symfony-console-completion/commit/c6fd5a2e9752f42379f7bcd58b2b6ae144bccd96

**Comment by duncan3dc** [comment]
@aik099 I don't care who else does it, what I care about is that using that technique has located a dependency issue. So what reason do you have to not use that that technique to detect issues in the future?

It's a _relatively_ new composer feature, so I don't expect it to have widespread usage yet anyway.

File: composer.json (hunk 1 above)
Comment (aik099) [inline composer.json]: I've reported #35 to replace usage of new PHPUnit method instead of increasing needed PHPUnit version.

**Comment by aik099** [comment]
> It's a relatively new composer feature, so I don't expect it to have widespread usage yet anyway.

This was Google says about `--prefer-lowest`: http://evertpot.com/testing-composer-prefer-lowest/

Also:
1. please restrict usage of `--prefer-lowest` to only PHP 5.4 to dramatically decrease total build count
2. adding another option to fail build if it fails on one of PHP version would also be useful

**Comment by duncan3dc** [comment]
@aik099 What's your point with that google link? It just seems to back me up so I'm not sure why you are posting it without comment

**Comment by duncan3dc** [comment]
@stecman Are you in agreement with @aik099? He seems to be speaking like a project collaborator here. If so I'll close this PR and be on my way. I was only trying to help you catch potential issues in your project as a thank you.

**Comment by aik099** [comment]
I'm posting a link so that anybody else stumbling on this issue in future won't need to find relevant info by himself.

I'm ok with improving Travis CI build, but I believe, that some work is still required on that `.travis.yml` file before it can be merged.

**Comment by stecman** [comment]
I'd probably opt for doing something like [what symfony/symfony does](https://github.com/symfony/symfony/blob/d4802aa92b15de24c7e30621b76cea444a20a64c/.travis.yml#L5-L12). Technically yes it's not testing all supported combinations, but testing all combinations feels a bit like it's just testing that the dependencies work with the PHP versions they claim to.

**Pull request merged**